### ISSUE TITLE
Use (infinite) streams in nushell plugins

### ISSUE THREAD

**Issue by Euphrasiologist** [nushell/nushell#6934]
### Related problem

The problem is that currently, nushell plugins accept only `nu_protocol::Value`s. I believe this is because the `Plugin` trait can only accept an input as a `Value`:

```rust
pub trait Plugin {
    fn signature(&self) -> Vec<Signature>;
    fn run(
        &mut self,
        name: &str,
        call: &EvaluatedCall,
        input: &Value,
    ) -> Result<Value, LabeledError>;
}
```

### Describe the solution you'd like

I'd like to make a plugin which accepts a stream, something like `PipelineData::ListStream` which is used in other core commands in nushell, which will be faster for very large files. I can only see a way to load the whole data structure into memory at the moment, as a `Vec<Value>` or similar.

### Describe alternatives you've considered

None yet.

### Additional context and details

I apologise if this is an easy fix or something I've overlooked!

**Comment by Euphrasiologist** [comment]
I don't know if this is a good first issue or not, but if I were to be given a little direction, I might be able to poke around in the code?

**Comment by rgwood** [comment]
Thanks for the issue, and sorry that nobody got back to you earlier.

Agreed that this is an unfortunate limitation in the current plugin system. There's been some discussion of how to do iterators in plugins in the #design-discussion channel on Discord: https://discord.com/channels/601130461678272522/615329862395101194/1049780878991179866

There are some ways we could accomplish it but they will likely be quite a bit of work. It's uncertain whether we will be able to deliver a feature like that before v1.0.

**Comment by sholderbach** [comment]
This has been implemented recently with #11911 by @devyn
